Codecademy/discourse-codebytes-plugin
# Fix save functionality for multi-line codebytes and add tests

## Overview
Fixes a bug where the contents of the CodeByte would not be updated in the Discourse editor if the text was more than one line. 
Also adds an acceptance test to catch this case in the future.

### PR Checklist
- [x] Related to JIRA ticket: [REACH-965](https://codecademy.atlassian.net/browse/REACH-965), [REACH-967](https://codecademy.atlassian.net/browse/REACH-967)

### Notes
This was an issue in the regex; `(.*)` matches any character _except linebreaks_, unless the `/s` flag is used.

### Testing Instructions
Automated testing:
- Run `rake plugin:qunit['CodeBytes']` from the Discourse directory

Manual testing:
1. Visit http://localhost:3000/
2. Create a new topic or a reply
3. In the composer on the left, insert a Codebyte using the button, then type multiple lines of code, e.g.
```
[codebyte]
for x in range(0, 4):
  print("Hello!")
[/codebyte]
```
4. In the CodeByte iframe on the right, select a language. Check that the [codebyte] tag on the left gets updated with the language attribute.
5. In the CodeByte iframe on the right, make some edits to the code. Click the "Save to post" button. Check that the codebyte contents on the left get updated with the new text.
6. In the CodeByte iframe on the right, make some edits to the code. Stop typing for a few seconds and wait for it to autosave. Check that the codebyte contents on the left get updated with the new text.
